### PR TITLE
[tune] Add documentation to --output flag

### LINKED ISSUE #4511: [tune] Add `--output` to the Tune docs
We should add --output to the docs.

_Originally posted by @richardliaw in https://github.com/ray-project/ray/pull/4322#issuecomment-477903993_

cc @andrewztan

### PR DESCRIPTION
**Pull request by andrewztan**
## What do these changes do?

Add documentation for the `--output` flag for ls / lsx in the Tune CLI.

## Related issue number

Closes #4511 

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by andrewztan** [comment]
@richardliaw

File: doc/source/tune-usage.rst [hunk 1]
```diff
@@ -490,12 +492,13 @@ Here are a few examples of command line calls.
     | MyTrainableClass | 3_height=54,width=21  | 7b99a28a   |
     +------------------+-----------------------+------------+
     Dropped columns: ['status', 'last_update_time']
+    Please increase your terminal size to view remaining columns.
 
-- ``tune list-experiments``: List tabular information about experiments within a project. Add the ``--sort`` flag to sort the output by specific columns. Add the ``--filter`` flag to filter the output in the format ``"<column> <operator> <value>"``.
+- ``tune list-experiments``: List tabular information about experiments within a project. Empty columns will be dropped by default. Additional columns may be dropped in order for the output to fit within the user's terminal window. Add the ``--sort`` flag to sort the output by specific columns. Add the ``--filter`` flag to filter the output in the format ``"<column> <operator> <value>"``. Add the ``--output`` flag to write the trial information to a specific file (CSV or Pickle).
```
Comment (richardliaw) [inline doc/source/tune-usage.rst]: Maybe we can be a little less verbose about the terminal viewing? The text in the example is already pretty clear; we can probably just keep `Empty columns will be dropped by default.` (for both `list-experiments` and `list-trials`).

**Review by richardliaw — CHANGES_REQUESTED** [review]

File: doc/source/tune-usage.rst (hunk 1 above)
Comment (andrewztan) [reply doc/source/tune-usage.rst]: Updated!

**Review by richardliaw — DISMISSED** [review]
Awesome, thanks!

**Comment by richardliaw** [comment]
BTW @andrewztan we should probably change the `output` not to go directly into the experiment directory:

```python
    if output:
        experiment_path = os.path.expanduser(experiment_path)
        output_path = os.path.join(experiment_path, output)
        file_extension = os.path.splitext(output)[1].lower()
        if file_extension in (".p", ".pkl", ".pickle"):
            checkpoints_df.to_pickle(output_path)
```

instead should probably just be:
```python
    if output:
        output_path = output
        file_extension = os.path.splitext(output)[1].lower()
        if file_extension in (".p", ".pkl", ".pickle"):
            checkpoints_df.to_pickle(output_path)
```

Does that make sense? If so, can we make this change and reflect this in the docs?

**Review by richardliaw — APPROVED** [review]

**Pull request merged**